### ISSUE TITLE
Same card is saved multiple times when paid via Google Pay

### ISSUE THREAD

**Issue by anant1811** [woocommerce/woocommerce-gateway-stripe#2327]
Reported in #4835572-zen

**Describe the bug**

The same card is saved multiple times when a subscription is paid via Google Pay:

![](https://uploads.linear.app/f629fcde-6f9b-48e8-8af3-60eed8823bff/f9d11f03-c4e3-4484-8ad5-8aec5792c514/14bbc7f8-4eaa-4d10-a22a-ea78477f061b?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2Y2MjlmY2RlLTZmOWItNDhlOC04YWYzLTYwZWVkODgyM2JmZi9mOWQxMWYwMy1jNGUzLTQ0ODQtOGFkNS04YWVjNTc5MmM1MTQvMTRiYmM3ZjgtNGVhYS00ZDEwLWEyMmEtZWE3ODQ3N2YwNjFiIiwiaWF0IjoxNzc0NzY0NDI3LCJleHAiOjE4MDYzMzQ5ODd9.wK4atci8yB7u05o6uLkdJa-rhBNEfEq129CvpVH6RaE)

Full Size: [https://d.pr/i/9ibIFk](<https://d.pr/i/9ibIFk>)

**To Reproduce**<br>Steps to reproduce the behavior:

1. Enable Payment Request buttons
2. Place an order for a subscription product and pay via Google Pay
3. Place another order, and pay again via Google Pay using the same card as used in step 2.
4. Check the saved cards, you'd see it's saved twice

**Expected behavior**

The same should be saved only once.

**Comment by rafaelzaleski** [comment]
I was able to reproduce it. The product needs to be a subscription. Otherwise, we don't save the payment method for express checkouts.

**Comment by chrismask** [comment]
Yes, is a fix coming? I know its priority is low but a fix is two lines of code

**Comment by rafaelzaleski** [comment]
> Yes, is a fix coming? I know its priority is low but a fix is two lines of code

Yes, it is in our backlog. If you already know how to solve it, you can submit the fix in a PR and assign me to review it.
